README.md

```diff
@@ -1388,27 +1388,148 @@ Then the log line will become:
 [2024-10-08T15:48:13Z INFO  topiary_core::tree_sitter] Processing match of query "comma spacing": LocalQueryMatch { pattern_index: 17, captures: [ {Node "," (1,3) - (1,4)} ] } at location (286,1)
 ```
 
+## Add a new language
+
+This section illustrates how to add a supported language to Topiary, provided it already has a tree-sitter grammar.
+
+We will use C as the running example in this section.
+
+### Minimal steps
+
+The two following steps are enough to jumpstart the formatting of a new language:
+
+#### Register the grammar in `topiary-config/languages.ncl`:
+
+```nickel
+    c = {
+      extensions = ["c", "h"],
+      grammar = {
+        git = "https://github.com/tree-sitter/tree-sitter-c.git",
+        rev = "6c7f459ddc0bcf78b615d3a3f4e8fed87b8b3b1b",
+      },
+    },
+```
+
+#### Create the query file
+```bash
+touch topiary-queries/queries/c.scm
+```
+
+#### Testing
+
+You can now check that Topiary is able to format your new language with:
```

Comment:
Maybe it would be worth adding scare quotes to manage the readers expectations :sweat_smile: 

```suggestion
You can now check that Topiary is able to "format" your new language with:
```

Maybe that's too subtle... What do you think?